core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1593,6 +1593,24 @@ trait ZIOFunctions extends Serializable {
    * current fiber for this operation to return non-empty lists.
    */
   final def children: UIO[IndexedSeq[Fiber[_, _]]] = descriptor.flatMap(_.children)
+
+  def fromOptionM[A](uio: UIO[Option[A]]): IO[Unit, A] = uio.flatMap {
+    case Some(a) => ZIO.succeed(a)
+    case None    => ZIO.fail(())
+  }
+
+  final def fromEitherE[E, A](io: Either[E, A]): IO[E, A] =
```

Comment:
The implementation is great and 100% correct. However, this function already exists inside `ZIOFunctions` trait, with the name `fromEither`.

However, good insight seeing that such a function could be useful to define `fromEitherM` (it is indeed quite useful).

Reply:
Ok, I'll not put this method on next commit. But there are some differences between  `fromEither` and ` fromEitherE`.